ci/images/ci-runner/hack/bin/deploy-cluster.sh

```diff
@@ -14,58 +14,93 @@ source "$SCRIPT_DIR/utils.sh"
     setx_on
 
-    echo "Wait until hypershift hosted cluster is ready..."
-    wait_period=0
-    while
-        [ \
-            "$(
-                kubectl -n clusters get hostedcluster "$CLUSTER_NAME" -o json \
-                    | jq -r '.status.version.history[0].state'
-            )" != "Completed" \
-        ]; do
-        if [ "$wait_period" -gt 1800 ]; then
-            echo "[ERROR] Failed to create OCP cluster." >&2
+    printf "Provision hypershift cluster...\n" | indent 2
+    rosa create cluster --cluster-name "$CLUSTER_NAME" \
+        --sts --mode=auto --oidc-config-id "24132snjs8gktc5rtufqv8hggjv3ivbt" \
+        --operator-roles-prefix plnsvc-ci --region "$REGION" --version "$OCP_VERSION" \
+        --compute-machine-type m5.2xlarge \
+        --subnet-ids="subnet-001487732ebdd14f4,subnet-0718fb663f4b97f38,subnet-0fe426997da62662c" \
+        --hosted-cp -y
+
+    printf "Wait till hypershift cluster is ready...\n" | indent 2
+    rosa logs install --cluster="$CLUSTER_NAME" --region "$REGION" --watch
+
+    printf "Hypershift is ready, create a cluster admin account for accessing the cluster\n" | indent 2
+    admin_output="$(rosa create admin --region "$REGION" --cluster="$CLUSTER_NAME")"
+ 
+    # Get the admin account credentials and API server URL
+    admin_user="$(echo "$admin_output" | grep -oP '(?<=--username ).*(?= --password)')"
+    admin_pass="$(echo "$admin_output" | grep -oP '(?<=--password ).*')"
+    api_url="$(echo "$admin_output" | grep -oP '(?<=oc login ).*(?= --username)')"
+
+    # Use the admin account to login to the cluster in a loop until the account is active.
+    max_retries=5
+    retries=0
+    export KUBECONFIG="$KUBECONFIG_DIR/kubeconfig"
+    while ! oc login "$api_url" --username "$admin_user" --password "$admin_pass" > /dev/null 2>&1; do
+        if [ "$retries" -eq "$max_retries" ]; then
+            echo "[ERROR] Failed to login the cluster." >&2
             print_debug_info
             exit 1
         fi
         sleep 60
-        wait_period=$((wait_period + 60))
-        echo "Waited $wait_period seconds..."
+        retries=$((retries + 1))
+        echo "Retried $retries times..."
     done
 
-    echo "Hypershift is ready, The following is Cluster credentials"
-    local pass
-    pass="$(kubectl get secret -n clusters "${CLUSTER_NAME}"-kubeadmin-password -o json | jq -r .data.password | base64 -d)"
-    echo "kubeadmin:${pass}"
-    echo "The following is the cluster kubeconfig"
+    printf "The following is the cluster kubeconfig:\n" | indent 2
+    cat "$KUBECONFIG"
 
-    hypershift create kubeconfig --name "$CLUSTER_NAME" | tee "$WORKSPACE/kubeconfig"
+    # Check all of clusteroperators and AVAILABLE are true
+    #Workaround: Check if apiserver is ready by calling kubectl get nodes
+    if ! timeout 300s bash -c "while ! kubectl get nodes >/dev/null 2>/dev/null; do printf '.'; sleep 10; done"; then
+        echo "API server is not ready"
```

Comment:
This needs to be redirected to `stderr`.

Reply:
fixed